test/CMakeLists.txt

```diff
@@ -19,8 +19,12 @@ if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
     set(XTL_INCLUDE_DIR ${xtl_INCLUDE_DIRS})
 endif ()
 
-message(STATUS "Forcing tests build type to Release")
-set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+if(NOT CMAKE_BUILD_TYPE)
+    message(STATUS "Setting tests build type to Release")
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+else()
+    message(STATUS "Tests build type is ${CMAKE_BUILD_TYPE}")
+endif()
```

Comment:
Actually we always run tests in Debug, benchmarks in Release mode. I find it comforting to run tests with additional sanity checks like out of bounds access etc. But I did not concern myself too much with xtensor Debug mode, is Debug not useful here? Or just slow?

Reply:
BTW, I think we did the same change some year ago in xtensor and xtensor-blas.

Reply:
Well, these are just for the tests of the XTL. We run the tests in Release mode since the Release behavior is different from the Debug one (especially regarding variable initialization). But indeed we did the same change in xtensor and it has proven to be useful, so let's merge it.